src/core/lib/security/transport/server_auth_filter.c

```diff
@@ -242,7 +242,6 @@ static grpc_error *init_channel_elem(grpc_exec_ctx *exec_ctx,
 
   GPR_ASSERT(!args->is_last);
   GPR_ASSERT(auth_context != NULL);
-  GPR_ASSERT(creds != NULL);
 
```

Comment:
Looking further I think it's fine as grpc_server_credentials_ref happily handles a NULL argument, and you added the check for chand->creds above.